pkg/helm/actions/uninstall_release.go

```diff
@@ -25,3 +30,24 @@ func UninstallRelease(name string, conf *action.Configuration) (*release.Uninsta
 
 	return resp, nil
 }
+
+func UninstallReleaseAsync(name string, ns string, version string, conf *action.Configuration, coreClient corev1client.CoreV1Interface) (*kv1.Secret, error) {
+	client := action.NewUninstall(conf)
+	go func() {
+		resp, err := client.Run(name)
+		if err != nil || resp == nil {
+			return
+		}
+
+		ch := resp.Release.Chart
+		if ch != nil && ch.Metadata != nil && ch.Metadata.Name != "" && ch.Metadata.Version != "" {
+			metrics.HandleconsoleHelmUninstallsTotal(ch.Metadata.Name, ch.Metadata.Version)
+		}
+	}()
+	secretName := fmt.Sprintf("sh.helm.release.v1.%v.v%v", name, version)
+	secret, err := coreClient.Secrets(ns).Get(context.TODO(), secretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, ErrReleaseNotFound
+	}
+	return secret, nil
```

Comment:
I believe there is no benefit from returning the Secret here. Here it's not wrong but in the HTTP response is not needed. Maybe you want keep it here for the unit tests? But its not required imho.

Reply:
Well in the sync uninstall api ,if the release does not exist or an incorrect name is passed for deletion, the response returned is ErrReleaseNotFound. So this logic i believe we can keep but yeah in case of successful deletion trigger let's not return the secret response.

Reply:
Yep, we should check for not found, but in the case of no error, or any other error should continue with the uninstallation, or?